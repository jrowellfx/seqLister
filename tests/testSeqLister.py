import seqLister

nonSeqList = []

print("Testing expandSeq()")
print(seqLister.expandSeq(1, nonSeqList))
print(seqLister.expandSeq(1))
print(seqLister.expandSeq([1, "004", 10, 15], nonSeqList))
print(seqLister.expandSeq(["1-4", "010-015"], nonSeqList))
print(seqLister.expandSeq(["1-10x2", "20-60x10"]))
print(seqLister.expandSeq(["5-1"], nonSeqList))
print(seqLister.expandSeq("5-1"))
print(seqLister.expandSeq(["5--2"], nonSeqList))
print(seqLister.expandSeq("5--2"))
print(seqLister.expandSeq(["10--10x2"], nonSeqList))
print(seqLister.expandSeq(["10--10x-2"], nonSeqList))
print(seqLister.expandSeq(["0-16x8", "0-16x2"], nonSeqList))
print(seqLister.expandSeq(["0-16x8", "0-16x2"]))
print(seqLister.expandSeq(["0-99x9"], nonSeqList))
print(seqLister.expandSeq("0-99x9"))
print(seqLister.expandSeq(["1-0100x9"], nonSeqList))
print(seqLister.expandSeq(["0-99x10"], nonSeqList))
print(seqLister.expandSeq(["0-9", "20-40x2"], nonSeqList))
print(seqLister.expandSeq(["0-6", "6-14x2", "14-70x10"], nonSeqList))
print(seqLister.expandSeq(["0-64x64", "0-64x32", "0-64x16", "0-64x8", "0-64x4", "0-64x2", "0-64"], nonSeqList))
print(seqLister.expandSeq(["-20--5"], nonSeqList))
print(seqLister.expandSeq(["-10--3"], nonSeqList))
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["1-6-12"], nonSeqList)) # invalid
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["1---6"], nonSeqList)) # invalid
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["1-6xa"], nonSeqList)) # invalid
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["1-6x2-"], nonSeqList)) # invalid
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["a-b"], nonSeqList)) # invalid
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["10--10x--2"], nonSeqList)) # invalid
print(nonSeqList)
nonSeqList = []
print(seqLister.expandSeq(["1-5", "8-a"], nonSeqList)) # partially invalid
print(nonSeqList)
nonSeqList = []

print()
print("Testing condenseSeq()")
print(seqLister.condenseSeq([]))
print(seqLister.condenseSeq([1]))
print(seqLister.condenseSeq(["2"], pad=4))
print(seqLister.condenseSeq([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
print(seqLister.condenseSeq([2, 1, 3, 7, 8, 4, 5, 6, 9, 10]))
print(seqLister.condenseSeq([0, 8, 16, 2, 4, 6, 10, 12, 14]))
print(seqLister.condenseSeq([0, 9, 18, 27, 36, 45, 54, 63, 72, 81, 90, 99]))
print(seqLister.condenseSeq([1, 10, 19, 28, 37, 46, 55, 64, 73, 82, 91, 100]))
print(seqLister.condenseSeq([0, 10, 20, 30, 40, 50, 60, 70, 80, 90]))
print(seqLister.condenseSeq([1, 1, 1]))
print(seqLister.condenseSeq([1, 1, 1, 2, 3]))
print(seqLister.condenseSeq([1, 1, 1, 3, 3, 5, 5, 5]))
print(seqLister.condenseSeq([1, 2]))
print(seqLister.condenseSeq([1, 5]))
print(seqLister.condenseSeq([1, 2, 5, 6, 9, 10]))
print(seqLister.condenseSeq([1, 5, 13]))

tmpList = seqLister.expandSeq(["0-100x2", 51], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["0-100x2", 51, 101, 102], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["1-5", "7-15x2"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["1-5", "7-15x2", "20-100x5"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["2-10x2", "14-30x4", "35-100x5"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

print(seqLister.condenseSeq([1, 2, 3, 4, 6, 8, 10]))
print(seqLister.condenseSeq([1, 2, 3, 4, 6, 8]))
print(seqLister.condenseSeq([1, 2, 3, 4, 6, 8, 10, 12, 13, 14, 15, 16]))
print(seqLister.condenseSeq([1, 2, 3, 4, 6, 8, 10, 12, 13, 14, 15]))
print(seqLister.condenseSeq([1, 2, 3, 4, 6, 8, 10, 11, 12]))
print(seqLister.condenseSeq([1, 2, 3, 4, 6, 8, 10, 11]))

print(seqLister.condenseSeq([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40]))
print(seqLister.condenseSeq([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 13, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40]))
print(seqLister.condenseSeq([0, 1, 2, 3, 4, 5, 6, 8, 10, 12, 14, 24, 34, 44, 54, 64]))
print(seqLister.condenseSeq([4, 5, 6, 8, 10, 12, 14, 24, 34, 44, 54, 64]))
print(seqLister.condenseSeq([0, 64, 32, 16, 48, 8, 24, 40, 56, 4, 12, 20, 28, 36, 44, 52, 60, 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62, 1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51], pad=2))
print(seqLister.condenseSeq([0, 64, 32, 16, 48, 8, 24, 40, 56, 4, 12, 20, 28, 36, 44, 52, 60, 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62, 1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21], pad=4))
print(seqLister.condenseSeq([0, 64, 32, 16, 48, 8, 24, 40, 56, 4, 12, 20, 28, 36, 44, 52, 60, 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42], pad=4))

tmpList = seqLister.expandSeq(["5-400x5", "7-400x7", "11-400x11", "13-400x13"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["5-400x5", "7-400x7", "11-400x11", "13-400x13", "17-400x17", "19-400x19"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["5-400x5", "7-400x7", "11-400x11", "13-400x13", "17-400x17", "19-400x19", "23-400x23", "29-400x29", "31-400x31"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["3-400x3", "5-400x5", "7-400x7", "11-400x11", "13-400x13", "17-400x17", "19-400x19", "23-400x23", "29-400x29", "31-400x31"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["2-400x2", "3-400x3", "5-400x5", "7-400x7", "11-400x11", "13-400x13", "17-400x17", "19-400x19", "23-400x23", "29-400x29", "31-400x31"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["2-400x2", "3-400x3", "5-400x5", "7-400x7", "11-400x11", "13-400x13", "17-400x17", "19-400x19", "23-400x23", "29-400x29", "31-400x31", "120-301"], nonSeqList)
print(seqLister.condenseSeq(tmpList))

tmpList = seqLister.expandSeq(["2-50x2", "3-50x3", "5-50x5", "7-50x7", "11-50x11", "13-50x13", "17-50x17", "19-50x19", "23-50x23"], nonSeqList)
print(seqLister.condenseSeq(tmpList))
