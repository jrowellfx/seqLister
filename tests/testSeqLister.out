Testing expandSeq()
[1]
[1]
[1, 4, 10, 15]
[1, 2, 3, 4, 10, 11, 12, 13, 14, 15]
[1, 3, 5, 7, 9, 20, 30, 40, 50, 60]
[5, 4, 3, 2, 1]
[5, 4, 3, 2, 1]
[5, 4, 3, 2, 1, 0, -1, -2]
[5, 4, 3, 2, 1, 0, -1, -2]
[10, 8, 6, 4, 2, 0, -2, -4, -6, -8, -10]
[10, 8, 6, 4, 2, 0, -2, -4, -6, -8, -10]
[0, 8, 16, 2, 4, 6, 10, 12, 14]
[0, 8, 16, 2, 4, 6, 10, 12, 14]
[0, 9, 18, 27, 36, 45, 54, 63, 72, 81, 90, 99]
[0, 9, 18, 27, 36, 45, 54, 63, 72, 81, 90, 99]
[1, 10, 19, 28, 37, 46, 55, 64, 73, 82, 91, 100]
[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40]
[0, 1, 2, 3, 4, 5, 6, 8, 10, 12, 14, 24, 34, 44, 54, 64]
[0, 64, 32, 16, 48, 8, 24, 40, 56, 4, 12, 20, 28, 36, 44, 52, 60, 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62, 1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63]
[-20, -19, -18, -17, -16, -15, -14, -13, -12, -11, -10, -9, -8, -7, -6, -5]
[-10, -9, -8, -7, -6, -5, -4, -3]
[]
[]
['1-6-12']
[]
['1---6']
[]
['1-6xa']
[]
['1-6x2-']
[]
['a-b']
[]
['10--10x--2']
[1, 2, 3, 4, 5]
['8-a']

Testing condenseSeq()
[]
['1']
['0002']
['1-10']
['1-10']
['0-16x2']
['0-99x9']
['1-100x9']
['0-90x10']
['1']
['1-3']
['1-5x2']
['1-2']
['1', '5']
['1-2', '5-6', '9-10']
['1', '5', '13']
['0-50x2', '51', '52-100x2']
['0-50x2', '51', '52-100x2', '101-102']
['1-4', '5-15x2']
['1-5', '7-13x2', '15-100x5']
['2-10x2', '14-26x4', '30-100x5']
['1-4', '6-10x2']
['1-4', '6', '8']
['1-4', '6-10x2', '12-16']
['1-3', '4-12x2', '13-15']
['1-4', '6', '8', '10-12']
['1-4', '6-10x2', '11']
['0-9', '20-40x2']
['0-9', '13', '20-40x2']
['0-6', '8-12x2', '14-64x10']
['4-5', '6-12x2', '14-64x10']
['00-52', '54-64x2']
['0000-0022', '0024-0064x2']
['0000-0044x2', '0048-0064x4']
['5', '7', '10-11', '13-15', '20-22', '25', '26-30x2', '33', '35', '39', '40-44x2', '45', '49-50', '52', '55-56', '60', '63', '65-66', '70', '75', '77-78', '80', '84-85', '88', '90-91', '95', '98-100', '104-105', '110', '112', '115', '117', '119-121', '125-126', '130', '132-133', '135', '140', '143-147x2', '150', '154-156', '160-161', '165', '168-170', '175-176', '180', '182', '185-189x2', '190', '195', '196-200x2', '203', '205', '208-210', '215', '217', '220-221', '224-225', '230-231', '234-235', '238-242x2', '245', '247', '250', '252-253', '255', '259-260', '264-266', '270', '273', '275', '280', '285-287', '290', '294-295', '297', '299-301', '305', '308-312x2', '315', '319-320', '322', '325', '329-330', '335', '336-340x2', '341-345x2', '350-352', '355', '357', '360', '363-365', '370-371', '374-375', '377-378', '380-390x5', '392', '395-396', '399-400']
['5', '7', '10-11', '13-15', '17', '19-22', '25', '26-30x2', '33-35', '38-40', '42', '44-45', '49-52', '55-57', '60', '63', '65', '66-70x2', '75-78', '80', '84-85', '88', '90-91', '95', '98-100', '102', '104-105', '110-114x2', '115', '117', '119-121', '125-126', '130', '132-133', '135-136', '140', '143-147x2', '150', '152-156', '160-161', '165', '168-171', '175-176', '180', '182', '185-189x2', '190', '195', '196-200x2', '203-205', '208-210', '215', '217', '220-221', '224-225', '228', '230-231', '234-235', '238-242x2', '245', '247', '250', '252-253', '255', '259-260', '264-266', '270', '272-273', '275', '280', '285-287', '289-290', '294-295', '297', '299-301', '304-305', '306-312x2', '315', '319-320', '322-323', '325', '329-330', '335-336', '338', '340-343', '345', '350-352', '355', '357', '360-361', '363-365', '370-371', '374-375', '377-378', '380', '385', '390-392', '395-396', '399-400']
['5', '7', '10-11', '13-15', '17', '19-23', '25-26', '28-31', '33-35', '38-40', '42', '44-46', '49-52', '55-58', '60', '62-63', '65-66', '68-70', '75-78', '80', '84-85', '87-88', '90-93', '95', '98-100', '102', '104-105', '110', '112', '114-117', '119-121', '124-126', '130', '132-133', '135', '136-140x2', '143-147x2', '150', '152-156', '160-161', '165', '168-171', '174-176', '180', '182', '184-187', '189-190', '195', '196-200x2', '203-205', '207-210', '215', '217', '220-221', '224-225', '228', '230-232', '234-235', '238-242x2', '245', '247', '248-252x2', '253', '255', '259-261', '264-266', '270', '272-273', '275-276', '279-280', '285-287', '289-290', '294-295', '297', '299-301', '304-305', '306-312x2', '315', '319-320', '322-323', '325', '329-330', '335-336', '338', '340-343', '345', '348', '350-352', '355', '357', '360-361', '363-365', '368', '370-372', '374-375', '377-378', '380', '385', '390-392', '395-396', '399-400']
['3', '5-7', '9-15', '17-31', '33-36', '38-40', '42', '44-46', '48-52', '54-58', '60', '62-63', '65-66', '68-70', '72', '75-78', '80-81', '84-85', '87-88', '90-93', '95-96', '98-100', '102', '104-105', '108', '110-112', '114-117', '119-121', '123-126', '129-130', '132-133', '135', '136-140x2', '141', '143-145', '147', '150', '152-156', '159-162', '165', '168-171', '174-177', '180', '182-187', '189-190', '192', '195', '196-200x2', '201', '203-205', '207-210', '213', '215-217', '219-222', '224-225', '228', '230-232', '234-235', '237', '238-242x2', '243', '245-250', '252-253', '255', '258-261', '264-267', '270', '272-273', '275-276', '279-280', '282', '285-291', '294-295', '297', '299-301', '303-306', '308-310', '312', '315', '318-325', '327', '329-330', '333', '335-336', '338-343', '345', '348', '350-352', '354-355', '357', '360-361', '363-366', '368-372', '374-375', '377-378', '380-381', '384-385', '387', '390-393', '395-396', '399-400']
['2-36', '38-40', '42', '44-46', '48-52', '54-58', '60', '62-66', '68-70', '72', '74-78', '80-82', '84-88', '90-96', '98-100', '102', '104-106', '108', '110-112', '114-126', '128-130', '132-136', '138', '140-148', '150', '152-156', '158-162', '164-166', '168-172', '174-178', '180', '182-190', '192', '194-196', '198', '200-210', '212-222', '224-226', '228', '230-232', '234-238', '240', '242-250', '252-256', '258-262', '264-268', '270', '272-276', '278-280', '282', '284-292', '294-306', '308-310', '312', '314-316', '318-330', '332-336', '338-346', '348', '350-352', '354-358', '360-366', '368-372', '374-378', '380-382', '384-388', '390-396', '398-400']
['2-36', '38-40', '42', '44-46', '48-52', '54-58', '60', '62-66', '68-70', '72', '74-78', '80-82', '84-88', '90-96', '98-100', '102', '104-106', '108', '110-112', '114-306', '308-310', '312', '314-316', '318-330', '332-336', '338-346', '348', '350-352', '354-358', '360-366', '368-372', '374-378', '380-382', '384-388', '390-396', '398-400']
['2-28', '30', '32-36', '38-40', '42', '44-46', '48-50']
